cpp/src/io/csv/csv_gpu.cu

```diff
@@ -570,6 +577,376 @@ __global__ void convertCsvToGdf(const char *raw_csv,
   }
 }
 
+/**
+ * @brief merge two packed row contexts (each corresponding to a block of characters)
+ * and returns the packed row context corresponding to the merged character block
+ **/
+inline __device__ packed_rowctx_t merge_row_contexts(packed_rowctx_t first_ctx,
+                                                     packed_rowctx_t second_ctx)
+{
+  uint32_t id0 = get_row_context(first_ctx, ROW_CTX_NONE) & 3;
+  uint32_t id1 = get_row_context(first_ctx, ROW_CTX_QUOTE) & 3;
+  uint32_t id2 = get_row_context(first_ctx, ROW_CTX_COMMENT) & 3;
+  return (first_ctx & ~pack_row_contexts(3, 3, 3)) +
+         pack_row_contexts(get_row_context(second_ctx, id0),
+                           get_row_context(second_ctx, id1),
+                           get_row_context(second_ctx, id2));
+}
+
+/**
```

Comment:
Low level functions should have doc blocks like this, but since they don't need to be in the public docs, don't start them with the doxygen indicator. (Suggest making this change to all the low-level functions in this PR -- we'll be working on documentation consistency across libcudf from here on.)

```suggestion
/*
```